### ISSUE TITLE
Mixing Dictionary styles is no longer possible

### ISSUE THREAD

**Issue by KoBeWi** [godotengine/godot#44977]
**Godot version:**

8158d17

**Issue description:**

```
var test = {
	some_value = 1,
	"other_value": 1
}
```
`Expected identifier as dictionary key.`
When you swap the keys:
`Expected ":" after dictionary key. Mixing dictionary styles is not allowed.`

If it's really not allowed, both errors should say so.

**Comment by KoBeWi** [comment]
It got worse in 97e041e (unrelated to #50057)

Pasting the above code results in editor crash:
```
[0] std::_Atomic_storage<unsigned int,4>::load (C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\atomic:935)
[1] std::_Atomic_storage<unsigned int,4>::load (C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\atomic:935)
[2] SafeNumeric<unsigned int>::conditional_increment (C:\godot_source\core\templates\safe_refcount.h:122)
[3] SafeRefCount::ref (C:\godot_source\core\templates\safe_refcount.h:169)
[4] StringName::StringName (C:\godot_source\core\string\string_name.cpp:151)
[5] Variant::Variant (C:\godot_source\core\variant\variant.cpp:2418)
[6] GDScriptParser::parse_dictionary (C:\godot_source\modules\gdscript\gdscript_parser.cpp:2500)
[7] GDScriptParser::parse_precedence (C:\godot_source\modules\gdscript\gdscript_parser.cpp:2015)
[8] GDScriptParser::parse_expression (C:\godot_source\modules\gdscript\gdscript_parser.cpp:2041)
[9] GDScriptParser::parse_variable (C:\godot_source\modules\gdscript\gdscript_parser.cpp:866)
[10] GDScriptParser::parse_variable (C:\godot_source\modules\gdscript\gdscript_parser.cpp:824)
[11] GDScriptParser::parse_class_member<GDScriptParser::VariableNode> (C:\godot_source\modules\gdscript\gdscript_parser.cpp:724)
[12] GDScriptParser::parse_class_body (C:\godot_source\modules\gdscript\gdscript_parser.cpp:778)
[13] GDScriptParser::parse_program (C:\godot_source\modules\gdscript\gdscript_parser.cpp:608)
[14] GDScriptParser::parse (C:\godot_source\modules\gdscript\gdscript_parser.cpp:388)
[15] GDScriptLanguage::lookup_code (C:\godot_source\modules\gdscript\gdscript_editor.cpp:2978)
[16] ScriptTextEditor::_validate_symbol (C:\godot_source\editor\plugins\script_text_editor.cpp:858)
[17] call_with_variant_args_helper<ScriptTextEditor,String const &,0> (C:\godot_source\core\variant\binder_common.h:218)
[18] call_with_variant_args<ScriptTextEditor,String const &> (C:\godot_source\core\variant\binder_common.h:333)
[19] CallableCustomMethodPointer<ScriptTextEditor,String const &>::call (C:\godot_source\core\object\callable_method_pointer.h:97)
[20] Callable::call (C:\godot_source\core\variant\callable.cpp:51)
[21] Object::emit_signal (C:\godot_source\core\object\object.cpp:1100)
[22] Object::emit_signal (C:\godot_source\core\object\object.cpp:1155)
[23] TextEdit::_gui_input (C:\godot_source\scene\gui\text_edit.cpp:2527)
[24] CodeEdit::_gui_input (C:\godot_source\scene\gui\code_edit.cpp:287)
[25] call_with_variant_args_helper<TextEdit,Ref<InputEvent> const &,0> (C:\godot_source\core\variant\binder_common.h:218)
[26] call_with_variant_args_dv<TextEdit,Ref<InputEvent> const &> (C:\godot_source\core\variant\binder_common.h:366)
[27] MethodBindT<TextEdit,Ref<InputEvent> const &>::call (C:\godot_source\core\object\method_bind.h:287)
[28] Viewport::_gui_call_input (C:\godot_source\scene\main\viewport.cpp:1656)
[29] Viewport::_gui_input_event (C:\godot_source\scene\main\viewport.cpp:2177)
[30] Viewport::input (C:\godot_source\scene\main\viewport.cpp:3065)
[31] Window::_window_input (C:\godot_source\scene\main\window.cpp:916)
[32] call_with_variant_args_helper<Window,Ref<InputEvent> const &,0> (C:\godot_source\core\variant\binder_common.h:218)
[33] call_with_variant_args<Window,Ref<InputEvent> const &> (C:\godot_source\core\variant\binder_common.h:333)
[34] CallableCustomMethodPointer<Window,Ref<InputEvent> const &>::call (C:\godot_source\core\object\callable_method_pointer.h:97)
[35] Callable::call (C:\godot_source\core\variant\callable.cpp:51)
[36] DisplayServerWindows::_dispatch_input_event (C:\godot_source\platform\windows\display_server_windows.cpp:1798)
[37] DisplayServerWindows::_dispatch_input_events (C:\godot_source\platform\windows\display_server_windows.cpp:1772)
[38] Input::_parse_input_event_impl (C:\godot_source\core\input\input.cpp:643)
[39] Input::parse_input_event (C:\godot_source\core\input\input.cpp:466)
[40] Input::flush_accumulated_events (C:\godot_source\core\input\input.cpp:859)
[41] DisplayServerWindows::process_events (C:\godot_source\platform\windows\display_server_windows.cpp:1534)
[42] OS_Windows::run (C:\godot_source\platform\windows\os_windows.cpp:622)
[43] widechar_main (C:\godot_source\platform\windows\godot_windows.cpp:163)
[44] _main (C:\godot_source\platform\windows\godot_windows.cpp:185)
[45] main (C:\godot_source\platform\windows\godot_windows.cpp:199)
[46] __scrt_common_main_seh (D:\a01\_work\26\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288)
[47] BaseThreadInitThunk
```

**Comment by KoBeWi** [comment]
The crash part is fixed by #51473

As for the original issue, mixing Dictionary styles is not allowed because _"that looks pretty awful to me, I don't see any reason to mix the styles"_ ~vnen 2021

After some discussion on chat, it was decided that LUA syntax should just allow quoted strings as keys. The reason I brought this up is that I prefer LUA style, but not all strings are valid identifiers, so allowing quoted strings would resolve this.

**Comment by JestemStefan** [comment]
If this is converted to discussion now then IMO mixing styles are confusing. 

Also from the perspective of beginner it can be confusing that one tutorial use one style and other use different style depends on what author like.

I would prefer if there was just 1 correct way to do this and then there will be no longer mixing of styles.

Also out of those two I prefer Python style so `{key:value}`
